z.go

```diff
@@ -39,3 +39,40 @@ func Dereference[T any](p *T) (v T) {
 	}
 	return
 }
+
+// AddToMap will add the key value pair to the map, ensuring that the map is not nil.
+func AddToMap[K comparable, V any](m map[K]V, k K, v V) map[K]V {
+	return ConcatMap(m, map[K]V{k: v})
+}
+
+// ConcatMap will add all the key value pairs from m2 to m1, overwriting any existing keys in m1.
+// If m1 is nil, then m2 is returned directly.
+func ConcatMap[K comparable, V any](m1 map[K]V, m2 map[K]V) map[K]V {
+	if m1 == nil {
+		return m2
+	}
+
+	for k, v := range m2 {
+		m1[k] = v
+	}
+
+	return m1
+}
+
+// ConcatMaps will iteratively add all the key/value pairs from each map, overwriting existing keys.
+// That is, if every map in ms has the same key, then the value of that key in the resulting map will be
+// the value of the key in the last map in ms.
+func ConcatMaps[K comparable, V any](ms ...map[K]V) map[K]V {
+	if len(ms) == 0 {
+		return nil
+	}
+	if len(ms) == 1 {
+		return ms[0]
+	}
+
+	for i := 1; i < len(ms); i++ {
+		ms[0] = ConcatMap(ms[0], ms[i])
+	}
+
+	return ms[0]
+}
```

Comment:
I don't think we need to expose `ConcatMap`, since calling `ConcatMaps` with two arguments does the same thing and is just as easy as calling `ConcatMap` would be.